plugin/src/main/java/com/dff/cordova/plugin/camera/helpers/CameraButtonHelper.java

```diff
@@ -0,0 +1,250 @@
+                captureRequest.set(CaptureRequest.CONTROL_AE_MODE,
+                                   CaptureRequest.CONTROL_AE_MODE_ON);
+                captureRequest.set(CaptureRequest.FLASH_MODE, CaptureRequest.FLASH_MODE_OFF);
+                flashMode = 2;
+                break;
+            case 2:
+                log.d(TAG, "auto flash");
+                flashButton.setImageResource(r.getDrawableIdentifier(IC_FLASH_AUTO));
+                captureRequest.set(CaptureRequest.CONTROL_AE_MODE,
+                                   CaptureRequest.CONTROL_AE_MODE_ON_AUTO_FLASH);
+                captureRequest.set(CaptureRequest.FLASH_MODE, CaptureRequest.FLASH_MODE_OFF);
+                flashMode = 0;
+                break;
+            default:
+                break;
+        }
+        
+    }
+    
+    /**
+     * Changes the icon of the button.
+     * @param flipMode state of the camera
+     */
+    public void changeFlipButton(CameraState flipMode) {
+        log.d(TAG, "changeCamera");
+        if (flipMode == CameraState.BACK) {
+            flipButton.setImageResource(r.getDrawableIdentifier(IC_CAMERA_BACK));
+        } else {
+            flipButton.setImageResource(r.getDrawableIdentifier(IC_CAMERA_FRONT));
+        }
+    }
+    
+    /**
+     * Enables or disables all buttons.
+     *
+     * @param enable true = enable, false = disable
+     */
+    public void enableAllButtons(boolean enable) {
+        flipButton.setEnabled(enable);
+        captureButton.setEnabled(enable);
+        flashButton.setEnabled(enable);
+    }
+    
+    /**
+     * Hides the button when there is no flashMode.
+     */
+    public void checkFlash() {
+        boolean hasFlashMode = packageManager.hasSystemFeature(PackageManager.FEATURE_CAMERA_FLASH);
+    
+        if (!hasFlashMode) {
+            log.d(TAG, "disable changeFlash button");
+            flashButton.setEnabled(false);
+            flashButton.setVisibility(View.GONE);
+        }
+    }
+    
+    /**
+     * Hides the button when the device does not have a front camera.
+     */
+    public void checkFrontCamera() {
+        boolean hasFrontCamera = packageManager
+            .hasSystemFeature(PackageManager.FEATURE_CAMERA_FRONT);
+    
+        if (!hasFrontCamera) {
+            log.d(TAG, "disable changeCamera button");
+            flipButton.setEnabled(false);
+            flipButton.setVisibility(View.GONE);
+        }
+    }
+    
+    /**
+     * The method changes between front and back camera.
+     */
+    public void changeCamera() {
+        if (cameraState == CameraState.FRONT) {
+            log.d(TAG, "flip to back camera");
+            try {
+                for (String id : cameraManager.getCameraIdList()) {
+                    CameraCharacteristics characteristics =
+                        cameraManager.getCameraCharacteristics(id);
+                    int cameraOrientation = characteristics.get(CameraCharacteristics.LENS_FACING);
```

Comment:
`Unboxing of 'characteristics.get(CameraCharacteristics.LENS_FACING)' may produce 'NullPointerException'`